client/rpc/src/eth.rs

```diff
@@ -813,8 +813,27 @@ impl<B, C, P, CT, BE, H: ExHashT> EthApiT for EthApi<B, C, P, CT, BE, H> where
 		}
 	}
 
+
+
 	fn estimate_gas(&self, request: CallRequest, _: Option<BlockNumber>) -> Result<U256> {
-		let calculate_gas_used = |request| -> Result<U256> {
+		let gas_limit = {
+            // query current block's gas limit
+			let hash = self.client.info().best_hash;
+            let block = self
+                .client
+                .runtime_api()
```

Comment:
We should try to use the overrides we have in place instead of creating new runtime api instances whenever possible. I.e. :
https://github.com/paritytech/frontier/blob/master/client/rpc/src/eth.rs#L254

Not critical as it's a single instance for this request, but still recommended to follow the rest of the handler implementations.

FYI this overrides are designed as optional, but in Moonbeam we have them enabled. They will read from storage instead of creating (and caching) a runtime instance to get the same data.

@crystalin which reminds me, once Substrate `frontier` branch is updated to latest `master`, we should run our perf test again now that the default block initialization is removed for runtime api calls.

Reply:
Thanks @tgmichel  for the suggestion, it's done :)